docs/monitoring.md

```diff
@@ -479,11 +479,27 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
     <td><code>/applications/[app-id]/stages/[stage-id]</code></td>
     <td>
       A list of all attempts for the given stage.
+        <br><code>?details=true</code> lists all attempts with the task data for the given stage.
+        <br><code>?withSummaries=true</code> lists task metrics distribution and executor metrics distribution of each attempt.
+        <br><code>?quantiles=0.1,0.25,0.5,0.75,1.0</code> summarize the metrics with the given quantiles. Query parameter quantiles takes effect only when <code>withSummaries=true</code>. Default value is <code>0.0,0.25,0.5,0.75,1.0</code>. 
```

Comment:
Is the 1.0 quantile useful? there's already a max in the API or is that not true.
I wonder if 0.9 or 0.95 is more useful

My only concern here is returning more data in the API but I suppose this is optional.

Reply:
The default quantiles value is 0.0,0.25,0.5,0.75,1.0 so that it is consistent with Spark web UI.   The quantile 1.0 value is useful to show the maximal metric value so that users can easily identify the bottleneck by using the REST call.  To my understanding, some downstream product such as LinkedIn/Dr. Elephant (https://github.com/linkedin/dr-elephant) computes the ratio of quantile-1.0-value/quantile-0.5-value to decide how the load is skewed/bottlenecked
among the tasks/executors of a stage.
If a user does not like the default value, he can always change it by explicitly specifying the quantile values he wants.

Reply:
> My only concern here is returning more data in the API but I suppose this is optional.

Yea, it's optional, but we can collect all data we need for each stage together instead of to call restful multiple times for different data.  and these data is also very useful for us to analysis spark jobs.